### DISCUSSION TITLE
Auto-index all fields in a JSON column

### DISCUSSION THREAD

**Discussion by criccomini** [duckdb/duckdb#5797] [Ideas]
I have a JSON column `my_json_column`. The column contains dictionaries with heterogenous schemas (i.e. there is no standard schema; different rows have different fields in `my_json_column`). I am querying using DuckDB's JSON path syntax:

```
SELECT * FROM table WHERE my_json_column->'$.location'->>'$.table' = 'managers';
```

It would be really great if I had a way to tell DuckDB to index all fields on all rows in the `my_json_column`. Essentially, similar behavior to what Elastic does. Then, my JSON path queries on `my_json_column` would be lightning fast, rather than requiring full scans.

**Comment by dforsber** [comment]
This sounds like a really nice feature! Point/search queries performance compared with Elastic is a good comparison point ;). If you reach that, the applicability of the db engine increases a lot. However, having big arbitrary indexes like Elastic requires hard work during data ingestion (as we know from Elastic) .. or during engine startup (really bad for embedded dbs to be slow on process start or db open). Anyway,  it's hard to beat Elastic in its own field (except with LIKE queries where it traditionally sucks (Lucene) - think they fairly recently added some additional indices (like all char triplets for string "contains" kind of queries), but still). In other words, you pay for the perf somewhere.
